### DISCUSSION TITLE
macOS: why is HomeBrew required to build libraries?

### DISCUSSION THREAD

**Discussion by imikejackson** [microsoft/vcpkg#30365] [Q&A] [answered]
I seem to get these errors a whole lot on my macOS machine:
```-- Fixing pkgconfig file: /opt/local/vcpkg/packages/sqlite3_x64-osx/lib/pkgconfig/sqlite3.pc
CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:619 (message):
  Could not find pkg-config.  Please install it via your package manager:

      brew install pkg-config
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:151 (vcpkg_find_acquire_program)
  ports/sqlite3/portfile.cmake:65 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:147 (include)


error: building sqlite3:x64-osx failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+sqlite3
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[sqlite3]+Build+error
Include '[sqlite3] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-12-14-7ae0d8527fb488fde10a89c2813802dc9b03b6f9
    vcpkg-scripts version: f14984af3 2023-01-09 (2 months ago)
```

By policy, I am not allowed to install HomeBrew and I'm not sure why this is necessary anyways? macOS does not rely on or need pkg-config.

I am just curious as to the thinking in requiring macOS developers to install a completely superfluous piece of software.

**Comment by Osyotr** [comment]
pkg-config is required by many ports to find their dependencies. It is also required by vcpkg to check whether .pc-files of a port are correct.
Brew is just one of the possible means to acquire pkg-config. If you manage to get it from some other place, you can avoid brew dependency.

**Reply by imikejackson** [reply]
But, it actually _isn't_ required to get their dependencies. We just get the port file and just change the cmake to something like "if(LINUX)  do package config stuff" and we don't have a problem.

Since vcpkg uses CMake, cmake will have those dependencies already figured out.

**Reply by Osyotr** [reply]
Could you please provide an example where `pkg-config` is redundant?

**Reply by dg0yt** [reply]
> Since vcpkg uses CMake, cmake will have those dependencies already figured out.

Are you sure you do have a sufficiently complete picture? Look at port libspatialite and ask yourself if cmake tells you which library names and flags to use with static linkage, for each possible feature combination. 
(pkg-config does this even for Windows, and it is actually used in vcpkg across *all* platforms to pass such information.)

**Comment by dg0yt** [comment, marked as answer]
> By policy, I am not allowed to install HomeBrew

You can do `vcpkg install pkgconf`. It is another name and implementation, but it provides what is needed.

**Reply by imikejackson** [reply]
What would be **REALLY** helpful would be for all those ReadMe's that just say install HomeBrew _just_ to get `pkgconf` might be updated or maybe some update to vcpkg cmake code to let macOS users know that they just simply need to `vcpkg install pkgconf` to get what is missing.

@dg0yt Thanks for the answer. This what I was looking for.
